TemsApi/src/Application/Managers/TypeManager.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Application.Common.Interfaces;
+using Domain.Entities;
+using Domain.Exceptions;
+
+namespace Application.Managers
+{
+    public class TypeManager
+    {
+        private readonly ITypeRepository _typeRepository;
+        public TypeManager(ITypeRepository typeRepository)
+        {
+            _typeRepository = typeRepository;
+        }
+
+        public async Task<Guid> CreateAsync(AssetType assetType, CancellationToken cancellationToken)
+        {
+            var existingTypes = await _typeRepository.FindByNameAsync(assetType.Name, cancellationToken);
+
+            if (existingTypes.Count() != 0) 
+                throw new ExistingTypeException("This type already exist!");
```

Comment:
Let's use this instead of throwing exceptions (https://www.youtube.com/watch?v=a1ye9eGTB98&t=704s)

Throwing exceptions is considered and anti-pattern.